tests/accessor/accessor_exceptions.hpp

```diff
@@ -0,0 +1,395 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  This file provides functions for tests on accessor exceptions.
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_GENERIC_ACCESSOR_PROPERTIES_H
+#define SYCL_CTS_GENERIC_ACCESSOR_PROPERTIES_H
+
+#include "../../util/usm_helper.h"
+#include "accessor_common.h"
+#include "catch2/catch_test_macros.hpp"
+
+#include "accessor_stub.hpp"
+
+#include <type_traits>
+
+namespace accessor_exceptions_test {
+
+// Class used as a kernel ID.
+template <accessor_tests_common::accessor_type, typename, int,
+          sycl_stub::target>
+struct Kernel;
+
+/**
+ * @brief Function helps to verify that constructors accessor really thrown
+ *        exceptions.
+ * @tparam AccType Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @tparam DataT Current data type
+ * @tparam Dimension Dimension size
+ * @tparam Target Current target
+ * @tparam GetAccFunctorT Type of the functor that will be use to interact with
+ *         accessor
+ */
+template <accessor_tests_common::accessor_type AccType, typename DataT,
+          int Dimension, sycl_stub::target Target, typename GetAccFunctorT>
+void check_exception(GetAccFunctorT construct_acc) {
+  auto queue = util::get_cts_object::queue();
+  DataT some_data(expected_val);
+  auto r = util::get_cts_object::range<Dimension>::get(1, 1, 1);
+  {
+    sycl::buffer<DataT, Dimension> data_buf(&some_data, r);
+
+    if constexpr (AccType == accessor_type::generic_accessor) {
+      auto action = [&] {
+        queue.submit([&](sycl::handler& cgh) { construct_acc(cgh, data_buf); })
+            .wait();
+      };
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::invalid));
+    } else if constexpr (AccType ==
+                         accessor_tests_common::accessor_type::host_accessor) {
+      auto action = [&] {
+        queue.submit([&](sycl::handler& cgh) { construct_acc(cgh, data_buf); })
+            .wait();
+      };
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::invalid));
+    } else if constexpr (AccType ==
+                         accessor_tests_common::accessor_type::local_accessor) {
+      auto action = [&] { construct_acc(queue); };
```

Comment:
I have felling that we can delete one branch here, by chaning condition to 
> `if constexpr (AccType != accessor_type::host_accessor)`
   {action for local and accessor}
`else`
   {action for host_accessor}

Reply:
We can't do it because we have differ parameters that should be provided into the `construct_acc` or we expect a differ exception. Therefore we should have three differ conditions